erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -265,6 +266,25 @@ def validate_reference_documents(self):
 						frappe.throw(_("{0} {1} must be submitted")
 							.format(d.reference_doctype, d.reference_name))
 
+	def validate_paid_invoices(self):
+		no_oustanding_refs = {}
+
+		for d in self.get("references"):
+			if not d.allocated_amount:
+				continue
+
+			if d.reference_doctype in ("Sales Invoice", "Purchase Invoice", "Fees"):
+				outstanding_amount = frappe.get_cached_value(d.reference_doctype, d.reference_name, "outstanding_amount")
```

Comment:
On selection of the sales invoice, system fetch the outstanding amount so no need to fetch that value again

Reply:
This is intentional. I have added a scenario to explain it.